### PR TITLE
wasm: Do not send a invalid ECDS response to Envoy when fetching WASM in the agent fails

### PR DESCRIPTION
**Pull request by ingwonsong**
This PR prevent to send the invalid ECDS, which doesn't have the required fields, to Envoy when fetching WASM in the agent fails.

Current implementation passes the ECDS to Envoy without any modification if fetching in the agent fails and `fail_open` is `true`.
But, if the ECDS does not have the required fields, it is rejected from Envoy and may cause errors in Envoy level.

### CODE REVIEW

File: pkg/wasm/convert.go [hunk 1]
```diff
@@ -151,7 +151,10 @@ func convert(resource *anypb.Any, cache Cache) (newExtensionConfig *anypb.Any, s
 	}
 	remote := vm.GetCode().GetRemote()
 	httpURI := remote.GetHttpUri()
-	if httpURI == nil {
+	if httpURI == nil || len(httpURI.GetUri()) == 0 {
+		// If the required fields are not set, do not send ECDS push to Envoy,
+		// because Envoy will reject it in the protocol validation.
+		sendNack = true
```
Comment (kyessenov) [inline pkg/wasm/convert.go:157]: So what happens to Envoy? Isn't it permanently un-initialized without ECDS response?

Comment (ingwonsong) [reply pkg/wasm/convert.go:157]: Yes. 

Do you think we need to try fetching Wasm somehow in Istio Agent?

**Comment by kyessenov** [comment]
I don't know. But it seems like whether or not this change is merged, Envoy
is still not initialized, so the PR is not accomplishing much.

**Comment by ingwonsong** [comment]
Ah..... Right. 

Then, what about filling such required fields with our best knowledge?

For example, 
- cluster_name: "PassthroughCluster"
- timeout: 15s

**Comment by ingwonsong** [comment]
BTW, close this PR, and will be back with the other way. Thanks!

**Comment by kyessenov** [comment]
I think we were talking about sending "empty" filters instead depending on the use case.
E.g. send "deny-all" filter for authz failed module and "allow-all" for telemetry modules.

**Pull request closed without merging**